brush-core/src/shell.rs

```diff
@@ -1715,22 +1724,32 @@ impl Shell {
     }
 }
 
+fn parse_impl<R: Read>(
+    r: R,
+    parser_options: brush_parser::ParserOptions,
+) -> Result<brush_parser::ast::Program, brush_parser::ParseError> {
+    let mut parser = create_parser(r, &parser_options);
+
+    tracing::debug!(target: trace_categories::PARSE, "Parsing reader as program...");
+    parser.parse_program()
+}
```

Comment:
Since there is only one caller of this function, you should inline this into `parse()`.

Reply:
Things like `parse_string` and `parse_string_impl` were separated before this before - do you think they should be inlined @reubeno/